File: kolibri-py/examples/handshake.py
"""sessionInit handshake against the real server, no SMS.

    python examples/handshake.py
"""

import kolibri

s = kolibri.Session("api.oneme.ru", 443)
print("state:", s.state())

info = s.connect()
print("state:", s.state())
print("calls_seed :", info["calls_seed"])
print("location   :", info["payload"].get("location"))
print("countries  :", len(info["payload"].get("reg-country-code", [])))

s.disconnect()
print("state:", s.state())

File: kolibri-dart/example/handshake.dart
import 'dart:io';

import 'package:kolibri/kolibri.dart';

Future<void> main() async {
  final libPath = Platform.environment['KOLIBRI_LIB'] ??
      'rust/target/debug/libkolibri_dart.dylib';
  await initKolibri(libraryPath: libPath);

  final session = openSession(host: 'api.oneme.ru');
  print('state: ${session.state()}');

  final info = await session.connect();
  print('state: ${session.state()}');
  print('calls_seed  : ${info.callsSeed}');
  print('device_name : ${info.deviceName}');
  print('payload     : ${info.payload.length} bytes of msgpack');

  session.disconnect();
  print('state: ${session.state()}');
}
